File: CodeShef/chef_and_meetings.py
test = input()
test = int(test)
for i in range(test):
    mtime = input()
    # intmtime = int(mtime[0:2]) + float(mtime[4:6])/100.0
    ampm = mtime[-2:]
    target = 0
    if (ampm == "AM"):
        if (int(mtime[0:2]) == 12):
            # print('adf')
            target += 0 + (float(mtime[3:5])/100.0)
        else:
            target += int(mtime[0:2]) + float(mtime[3:5])/100.0
    else:
        if (int(mtime[0:2]) == 12):
            target += int(mtime[0:2]) + float(mtime[3:5])/100.0
        else:
            target += 12 + int(mtime[0:2]) + float(mtime[3:5])/100.0
    # print(target)
    n = input()
    n = int(n)
    ans = ""
    for j in range(n):
        between = input()
        b1 = between[6:8]
        b2 = between[-2:]
        # print("b1", b1)
        # print('b2', b2)
        first = 0
        second = 0
        if (b1 == "AM"):
            if (int(between[0:2]) == 12):
                first += 0 + float(between[3:5])/100.0
            else:
                first += int(between[0:2]) + float(between[3:5])/100.0
        else:
            if (int(between[0:2]) == 12):
                first += int(between[0:2]) + float(between[3:5])/100.0
            else:
                first += 12 + int(between[0:2]) + float(between[3:5])/100.0
        if (b2 == "AM"):
            if (int(between[9:11]) == 12):
                second += 0 + float(between[12:14])/100.0
            else:
                second += int(between[9:11]) + float(between[12:14])/100.0
        else:
            if (int(between[9:11]) == 12):
                second += int(between[9:11]) + float(between[12:14])/100.0
            else:
                second += 12 + int(between[9:11]) + float(between[12:14])/100.0
        # print("first ", first)
        # print('second ', second)
        if target >= first and target <= second:
            ans += '1'
        else:
            ans += '0'
    print(ans)

File: CodeShef/chef_and_meetings.cpp
#include<iostream>
#include<string>
using namespace std;
int main(){
    int test;
    cin >> test;
    while(test--){
        string a;
        string b;
        cin >> a; 
        cin >> b;
        int e = stoi(a.substr(0,2));
        double f = stoi(a.substr(3,5))/100.00;
        double target = 0;
        if (b == "AM"){
            if (e == 12){
                target += 0 + f;
            }else{
                target += e + f;
            }
        }else{
            if (e == 12){
                target += e + f;
            }else{
                target += 12 + (e + f);
            }
        }
        // cout << target<<endl;
        int n = 0;
        cout << target <<endl;
        cin >> n;
        string ans = "";
        while(n--){
            string s;
            cin >> s;
            string s2;
            string s3;
            string s4;
            cin >> s2;
            cin >> s3;
            cin  >> s4;
            // cout << s << endl;
            // cout << s2 << endl;
            // cout << s3 << endl;
            // cout << s4 << endl;
            int t1 = stoi(s.substr(0,2));
            double t2 = stoi(s.substr(3,5))/100.00;
            int t3 = stoi(s3.substr(0,2));
            double t4 = stoi(s3.substr(3,5))/100.00;
            // cout << t1 << endl;
            // cout << t2 << endl;
            // cout << t3 << endl;
            // cout << t4 << endl;
            double start = 0;
            double end = 0;
            if (s2 == "AM"){
                if (t1 == 12){
                    start = 0 + t2;
                }else{
                start += t1 + t2;
                }
            }else{
                start += 12 +(t1 + t2);
                if (t1 == 12){
                    start -= 12;
                }
            }
            if (s4 == "AM"){
                if (t3 == 12){
                    end = 0 + t4;
                }else{
                end += t3 + t4;
                }
            }else{
                end = 12 + (t3 + t4);
                if (t3 == 12){
                    end -= 12;
                }
            }
            cout << start << endl;
            cout << end << endl;
            if (target >= start && target <= end){
                ans += '1';
            }else{
                ans += '0';
            }


        }
        cout << ans << endl;
    }
}